CONTRIBUTING.md

```diff
@@ -0,0 +1,227 @@
+# Contributing
+The following guide will walk you through contributing new pipelines or changes to existing pipelines and contains a troubleshooting section. Please also read [DISTRIBUTION.md](docs/DISTRIBUTION.md) to understand how our pipelines are distributed to the users.
+
+## Walktrough: Prerequesites
+
+To start development in the pipelines repository, there are a few steps you need to do to ensure you have the setup.
+
+
+### 1. Prepare the repository
+
+1. Fork the [pipelines]() repository on github, alternative check out this repository if you have the right to create Pull Requests directly.
+2. Clone the forked repository
+```sh
+git clone git@github.com:rudolfix/pipelines.git
+```
+3. Make a feature branch in the fork
+```sh
+cd pipelines
+git checkout -b <your-branch-name>
+```
+
+### 2. Prepare the development environment
+Development on the pipelines repository depends on python 3.8 or higher and poetry being available as well as the needed dependencies being installed. Make is used to automate tasks
+
+1. Install poetry:
+```sh
+make install-poetry
+```
+2. Install the python dependencies:
+```sh
+make dev
+```
+3. Activate the poetry shell:
+```sh
+poetry shell
+```
+4. To verify that the dependencies are set up correctly you can run the linter / formatter:
+```sh
+make format-lint
+```
+If this command fails, something is not set up correctly yet.
+
+
+## Walktrough: Create and contribute a new pipeline
+In this section you will learn how to contribute a new pipeline based on the `chess` pipeline. The `chess` pipeline is a very basic pipeline with two sources and a few resources. For an even simpler starting point, you can use the `pokemon` pipeline as the starting point with the same method. Please also read [DISTRIBUTION.md](docs/DISTRIBUTION.md) before you start this guide to get an understand of how your pipeline will be distributed to other users once it is accepted into our repo.
+
+1. Ensure you have followed all steps in the prerequesites section.
+2. We will copy the chess pipeline as a starting point. You can copy any other pipeline you wish to serve as your starting point. Another fairly simple pipeline is the chess pipeline for example. We will assume your new pipeline is called `animals`.
```

Comment:
Please fix formatting

Reply:
I checked the files in vscode preview and made some small changes, now it should be good. I can check after we merge on GitHub and fix remaining stuff (or is there a way to see it now already?)